ember-headless-form/src/components/-private/field.ts

```diff
@@ -0,0 +1,45 @@
+import Component from '@glimmer/component';
+import { HeadlessFormData } from '../headless-form';
+import LabelComponent from './field/label';
+import InputComponent from './control/input';
+import { WithBoundArgs } from '@glint/template';
+
+export interface HeadlessFormFieldComponentSignature<
+  DATA extends HeadlessFormData
+> {
+  Args: {
+    data: Partial<DATA>;
+    name: keyof DATA;
+  };
+  Blocks: {
+    default: [
+      {
+        label: WithBoundArgs<typeof LabelComponent, 'fieldId'>;
+        input: WithBoundArgs<typeof InputComponent, 'fieldId' | 'value'>;
+      }
+    ];
+  };
+}
+
+export default class HeadlessFormFieldComponent<
+  DATA extends HeadlessFormData
+> extends Component<HeadlessFormFieldComponentSignature<DATA>> {
+  LabelComponent = LabelComponent;
+  InputComponent = InputComponent;
+
+  get value(): DATA[keyof DATA] | undefined {
+    return this.args.data[this.args.name];
+  }
+
+  // copy-pasted from https://github.com/emberjs/ember.js/blob/master/packages/@ember/-internals/glimmer/lib/helpers/unique-id.ts
+  // we can remove this once we only support Ember 4.4+!
```

Comment:
anything wrong with only supporting Ember 4.4+?

Reply:
Hm, good question. No, probably not? 🤷‍♂️ At least for a fresh new addon...

Reply:
Only supporting Ember 4.4+, and using native `{{unique-id}}`